### PR TITLE
BAU: ConfigurationService Consistency Improvements

### PR DESCRIPTION
**Pull request by BenjaminWCO**
## What

- Added helper functions  in Config Service for common patterns for flags and URIs.
- Config Service consistently returns URI objects rather than a mix of strings and URIs.
- URI environment variables no longer default to empty when when missing -will throw instead - this behaviour seemed to be relied on in tests only.
- More consistent method names in Config Service.

## How to review

Code Review

## Checklist

- [x] Impact on orch and auth mutual dependencies has been checked.

### CODE REVIEW

**Comment by hannah-leong** [comment]
Just had a quick look but finding it quite difficult to review - would it be possible to split it up into smaller commits/PRs? (Also sonar is a lil grumpy)

**Comment by BenjaminWCO** [comment]
> Just had a quick look but finding it quite difficult to review - would it be possible to split it up into smaller commits/PRs? (Also sonar is a lil grumpy)

Sure, I'll make separate PRs just with the renaming and helper functions in the ConfigService and rebase this one once that's merged. That should make any other changes a little easier to parse. I think Sonar is grumpy because I changed the return type on getOidcApiBaseURL in the config service and there were already a few places where `Optional.get()` that was called without verifying if it was present. I actually remove this behaviour in a PR that is dependant on this one: https://github.com/govuk-one-login/authentication-api/pull/4473/files#diff-ecd2c579cef530e0394277d380a195f8ec7ff74bb7a6c6bdece4267a9633e93a. The low code coverage, I suspect, is just the result of making a lot of changes in the configuration service.

File: oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandlerTest.java [hunk 1]
```diff
@@ -162,6 +162,8 @@ class AuthorisationHandlerTest {
     private static final String EXPECTED_LANGUAGE_COOKIE_STRING =
             "lng=en; Max-Age=31536000; Domain=auth.ida.digital.cabinet-office.gov.uk; Secure; HttpOnly;";
     private static final URI LOGIN_URL = URI.create("https://example.com");
+    private static final URI INTERNAL_SECTORD_URI =
```
Comment (isaac-GDS) [inline oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandlerTest.java:165]: syntax error - `SECTORD` should be `SECTOR`

File: oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandlerTest.java [hunk 2]
```diff
@@ -221,6 +223,7 @@ public void setUp() {
         when(configService.getSessionCookieAttributes()).thenReturn("Secure; HttpOnly;");
         when(configService.getSessionCookieMaxAge()).thenReturn(3600);
         when(configService.getPersistentCookieMaxAge()).thenReturn(34190000);
+        when(configService.getInternalSectorURI()).thenReturn(INTERNAL_SECTORD_URI);
```
Comment (isaac-GDS) [inline oidc-api/src/test/java/uk/gov/di/authentication/oidc/lambda/AuthorisationHandlerTest.java:226]: syntax error - Same as the above

**Comment by ethanmills** [comment]
Please update the commit message and PR description to describe what changes you have made and why.

**Pull request closed without merging**